src/service/viewer-impl.js

```diff
@@ -802,6 +802,25 @@ export class Viewer {
   }
 
   /**
+   * Get the fragment from the url or the viewer.
+   * Strip leading '#' in the fragment
+   * @return {!Promise<string|undefined>}
+   */
+  getFragment() {
+    if (!this.isEmbedded_) {
```

Comment:
I think we can only get the share-tracking info from search viewer using message channel, am I understanding it correctly?